app/src/main/java/com/safetyheads/akademiaandroida/YouTube/repository/ChannelRepositoryImpl.kt

```diff
@@ -0,0 +1,29 @@
+package com.safetyheads.akademiaandroida.YouTube.repository
+
+import com.safetyheads.akademiaandroida.YouTube.entities.chanel.ChannelDataClass
+import com.safetyheads.akademiaandroida.network.ApiClient
+import com.safetyheads.akademiaandroida.network.NetworkResult
+import com.safetyheads.akademiaandroida.network.YouTubeApi
+import com.safetyheads.akademiaandroida.network.YouTubeService
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+
+class ChannelRepositoryImpl : ChannelRepository {
+
+    override suspend fun getYouTubeChannel(): Flow<NetworkResult<ChannelDataClass>> =
+        flow {
+            val retrofitYouTubeChannel =
+                ApiClient().create(YouTubeApi.YOUTUBE_API_BASE_URL, YouTubeService::class.java)
```

Comment:
YoutubeService should be provided via constructror:
https://www.samouczekprogramisty.pl/dependency-inversion-principle-dependency-injection-i-inversion-of-control/